ark/utils/load_utils.py

```diff
@@ -249,85 +177,133 @@ def load_imgs_from_tree(data_dir, img_sub_folder=None, fovs=None, channels=None,
+                 of channels in the input.
     """
+    if not os.path.isdir(data_dir):
+        raise ValueError(f"Invalid value for data_dir. {data_dir} is not a directory.")
 
-    imgs = iou.list_files(data_dir, substrs=['.tif', '.jpg', '.png'])
-
-    imgs.sort()
+    if files is None:
+        imgs = iou.list_files(data_dir, substrs=['.tif', '.jpg', '.png'])
+    else:
+        imgs = files
+        for img in imgs:
+            if not os.path.isfile(os.path.join(data_dir, img)):
+                raise ValueError(f"Invalid value for {img}. "
+                                 f"{os.path.join(data_dir, img)} is not a file.")
 
     if len(imgs) == 0:
         raise ValueError(f"No images found in directory, {data_dir}")
 
     test_img = io.imread(os.path.join(data_dir, imgs[0]))
 
+    # check data format
+    multitiff = test_img.ndim == 3
+    channels_first = multitiff and test_img.shape[0] == min(test_img.shape)
+
+    # check to make sure all channel indices are valid given the shape of the image
+    # and that channels_names has the same length as the number of channels in the image
+    n_channels = 1
+    if multitiff:
+        n_channels = test_img.shape[0] if channels_first else test_img.shape[2]
+        if channel_indices:
+            if max(channel_indices) >= n_channels or min(channel_indices) < 0:
+                raise ValueError(f'Invalid value for channel_indices. Indices should be'
+                                 f' between 0-{n_channels-1} for the given data.')
+    if xr_channel_names and n_channels != len(xr_channel_names):
+        raise ValueError(f'Invalid value for xr_channel_names. xr_channel_names'
+                         f' length should be {n_channels}, as the number of channels'
+                         f' in the input data.')
+
     # check to make sure that float dtype was supplied if image data is float
     data_dtype = test_img.dtype
     if force_ints and np.issubdtype(dtype, np.integer):
         if not np.issubdtype(data_dtype, np.integer):
-            warnings.warn(f"The the loaded {data_dtype} images were forcefully "
+            warnings.warn(f"The loaded {data_dtype} images were forcefully "
                           f"overwritten with the supplied integer dtype {dtype}")
     elif np.issubdtype(data_dtype, np.floating):
         if not np.issubdtype(dtype, np.floating):
             warnings.warn(f"The supplied non-float dtype {dtype} was overwritten to {data_dtype}, "
                           f"because the loaded images are floats")
             dtype = data_dtype
 
-    if variable_sizes:
-        img_data = np.zeros((len(imgs), 1024, 1024, 1), dtype=dtype)
-    else:
-        img_data = np.zeros((len(imgs), test_img.shape[0], test_img.shape[1], 1),
-                            dtype=dtype)
+    # extract data
+    img_data = []
+    for img in imgs:
+        v = io.imread(os.path.join(data_dir, img))
+        if not multitiff:
+            v = v[:, :, np.newaxis]
```

Comment:
There's nothing wrong with this approach, but I think using `np.expand_dims` is more standard and a bit easier to read

Reply:
I changed it